### DISCUSSION TITLE
MvtLayer with ArcGIS Vector Tile Server?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#5482] [Q&A]
Hi! Looking for some guidance on how to best go about using the MvtLayer with an ArcGIS Vector Tile Server. Would there be any examples or docs available?

**Comment by ibgreen** [comment]
@jfuehner I am not aware that anyone has been testing this, but it is definitely a use case that we want to support.

We do regularly load I3S models from public ArcGIS URLs and it pretty much just works. 

Do you have any public URLs we can test with, or it is an authentication issue related to private tiles?

**Comment by jfuehner** [comment]
@ibgreen all of the Arc services I deal with are private servers...

However, I took the MvtLayer [example](https://deck.gl/docs/api-reference/geo-layers/mvt-layer) from the deck.gl website and created a [CodePen](https://codepen.io/jfuehner/pen/GRNrbOE) where I tried to use an ArcGIS Vector Tile public [service](https://basemaps.arcgis.com/arcgis/rest/services/World_Basemap_Export_v2/VectorTileServer) I found available online to do some testing with. The MvtLayer not appear to handle the protobuf format being returned...

I guess I was more so looking for insights into how best to use the MvtLayer with the Arc Vector Services as that does not seem to be the standard use-case... Does the [I3SLoader](https://loaders.gl/modules/i3s/docs/api-reference/i3s-loader) handle pbf format? I see it handles bin/binary format but wasn't sure if that included pbf's...

**Reply by athomann** [reply]
Were you able to get your ArcGIS Vector Tile to not appear black like in your codepen?

**Comment by ibgreen** [comment]
> The MvtLayer not appear to handle the protobuf format being returned...

Mapbox Vector Tiles are always `pbf` encoded according to the mapbox specification. Unless Esri are storing some variant geometry that we have missed, it should just work. Note that there was a regression in the first 8.4 releases that has been fixed, if you are using 8.4 you may want to make sure you are on the latest version of deck.gl.


> Does the I3SLoader handle pbf format? I see it handles bin/binary format but wasn't sure if that included pbf's...

No the i3SLoader implements the I3S OGC standard (created by Esri) which is not based on pbf.

**Comment by jfuehner** [comment]
@ibgreen thanks for the information about the 8.4 regression issue. looks like with this [CodePen](https://codepen.io/jfuehner/pen/GRNrbOE) I am using the latest deck; v8.4.5. I receive the below error...

![image](https://user-images.githubusercontent.com/1843941/107886489-159f1e80-6ece-11eb-8449-9832b7082733.png)

When I roll back to a pre-8.4 version (i.e., 8.3.15) with this [CodePen](https://codepen.io/jfuehner/pen/LYbxwjz) the tiles are loading as expected...

**Reply by ibgreen** [reply]
Yes this seems like the real issue. I suspect this is related to the more flexible loader selection system in 8.4. 

The `MVTLoader` in loaders.gl currently recognizes the following MIME types: `application/x-protobuf`, `application/vnd.mapbox-vector-tile` and a '.mvt' extension. But currently not `application/octet-stream` and `.pbf` We can certainly add them though they are quite generic and could match other registered loaders. You are welcome to put up a PR in loaders.gl for discussion.

To unblock, you could probably use the new `Layer.loaders` prop to force the loader to `MVTLoader`.

@alasarr FYI

**Reply by kylebarron** [reply]
If we accept the protobuf media type we might as well also accept the `.pbf` extension

**Reply by ibgreen** [reply]
https://github.com/visgl/loaders.gl/pull/1180

**Comment by Pessimistress** [comment]
This should be fixed in 8.4.7.
